homeassistant/runner.py

```diff
@@ -48,34 +48,38 @@ def __init__(self, debug: bool) -> None:
     @property
     def loop_name(self) -> str:
         """Return name of the loop."""
-        return self._loop_factory.__name__
+        return self._loop_factory.__name__  # type: ignore
 
-    def new_event_loop(self):
+    def new_event_loop(self) -> asyncio.AbstractEventLoop:
         """Get the event loop."""
-        loop = super().new_event_loop()
+        loop: asyncio.AbstractEventLoop = super().new_event_loop()
         loop.set_exception_handler(_async_loop_exception_handler)
         if self.debug:
             loop.set_debug(True)
 
         executor = ThreadPoolExecutor(thread_name_prefix="SyncWorker")
         loop.set_default_executor(executor)
-        loop.set_default_executor = warn_use(  # type: ignore
-            loop.set_default_executor, "sets default executor on the event loop"
+        setattr(
+            loop,
+            "set_default_executor",
+            warn_use(
+                loop.set_default_executor, "sets default executor on the event loop"
+            ),
```

Comment:
Why is this needed? Looks really ugly.

Reply:
Ugliness is subjective, but it's because of https://github.com/python/mypy/issues/2427, better than `# type: ignore`.

Reply:
🤷 I'm fine either way.

Reply:
`setattr` not triggering the same errors as assigning to a method might be fixed one day in mypy though 😉

Reply:
Yeah, it could be fixed either way in mypy, or perhaps there are several related things to fix. Anyway it's good to construct our workarounds in away that they start to trigger errors when they're no longer needed, so we get to clean them up as soon as possible. And at least we know unneeded type ignores result in that, so from that POV that instead of a setattr would be preferable after all -- and at least it's more prominent. Will have a look.

Reply:
Changed, passes locally, let's see if CI passes. Some headscratching here because the above discussed item already had a type ignore, not sure why I ended touching it in the first place.